include/communication/EmscriptenCommunication.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2024 UltiMaker
+// CuraEngine is released under the terms of the AGPLv3 or higher
+
+#ifndef EMSCRIPTENCOMMUNICATION_H
+#define EMSCRIPTENCOMMUNICATION_H
+#ifdef __EMSCRIPTEN__
+
+#include "communication/CommandLine.h"
+
+namespace cura
+{
+
+/**
+ * \class EmscriptenCommunication
+ * \brief A class for handling communication in an Emscripten environment.
+ *
+ * This class extends the CommandLine class and provides specific implementations
+ * for sending progress and handling slice information in an Emscripten environment.
+ */
+class EmscriptenCommunication : public CommandLine
+{
+private:
+    std::string progress_handler_; ///< Handler for progress messages.
+    std::string gcode_prefix_handler_; /// <Handler for getting gcode_prefix.
```

Comment:
The "prefix" naming is a bit confusing, if I'm correct we mostly use "header" for this, in the front-end and in the engine (although it seems inconsistent in the engine yet). Would you mind renaming the methods/variables to "header" ?